x11-wm/fvwm3/fvwm3-1.0.4.ebuild

```diff
@@ -0,0 +1,195 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{8,9,10,11} )
+GO_OPTIONAL=1
+inherit autotools desktop flag-o-matic go-module python-single-r1
+
+DESCRIPTION="A multiple large virtual desktop window manager derived from fvwm"
+HOMEPAGE="http://www.fvwm.org/"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/fvwmorg/fvwm3.git"
+	EGIT_BRANCH="master"
+else
+	SRC_URI="https://github.com/fvwmorg/fvwm3/releases/download/${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+SRC_URI+=" https://gitlab.com/Matt.Jolly/fvwm3-go-deps/-/raw/${PV}/${P}-deps.tar.xz?inline=false -> ${P}-deps.tar.xz"
+
+LICENSE="GPL-2 FVWM"
+SLOT="0"
+
+IUSE="bidi debug doc go htmldoc netpbm nls perl png readline rplay stroke svg tk truetype vanilla lock x86-fbsd"
+REQUIRED_USE="
+	htmldoc? ( doc )
+	${PYTHON_REQUIRED_USE}"
+
+DOCS=( NEWS )
+
+if [[ ${PV} == 9999 ]]; then
+	DOCS+=( dev-docs/COMMANDS dev-docs/DEVELOPERS.md dev-docs/INSTALL.md dev-docs/PARSING.md dev-docs/TODO.md dev-docs/NEW-COMMANDS.md )
+fi
+
+BDEPEND="
+	app-arch/unzip
+	sys-libs/zlib
+	x11-libs/libICE
+	x11-libs/libSM
+	x11-libs/libX11
+	x11-libs/libXau
+	x11-libs/libxcb
+	x11-libs/libXcursor
+	x11-libs/libXdmcp
+	x11-libs/libXext
+	x11-libs/libXfixes
+	x11-libs/libXpm
+	x11-libs/libXrandr
+	x11-libs/libXrender
+	bidi? ( dev-libs/fribidi )
+	go? ( >=dev-lang/go-1.14 )
+	png? ( media-libs/libpng )
+	readline? (
+		sys-libs/ncurses
+		sys-libs/readline
+	)
+	stroke? ( dev-libs/libstroke )
+	svg? ( gnome-base/librsvg )
+	truetype? (
+		media-libs/fontconfig
+		x11-libs/libXft
+	)"
+
+RDEPEND="${PYTHON_DEPS}
+	${COMMON_DEPEND}
+	!x11-wm/fvwm
+	dev-lang/perl
+	perl? ( tk? (
+			dev-lang/tk
+			dev-perl/Tk
+			>=dev-perl/X11-Protocol-0.56
+		)
+	)
+	rplay? ( media-sound/rplay )
+	lock? ( x11-misc/xlockmore )
+	userland_GNU? ( sys-apps/debianutils )
+	!x86-fbsd? ( netpbm? ( media-libs/netpbm ) )"
+
+DEPEND="${COMMON_DEPEND}
+	virtual/pkgconfig
+	doc? ( dev-libs/libxslt
+		dev-ruby/asciidoctor )
+	x11-base/xorg-proto"
+
+src_unpack() {
+	if [[ ${PV} == 9999 ]]; then
+		einfo "The branch ${EGIT_BRANCH} will be installed."
+		git-r3_src_unpack
+	else
+		unpack "${P}".tar.gz
+	fi
+	unpack "${P}"-deps.tar.xz
+	mv go-mod ${P}/bin/FvwmPrompt/
+}
+
+PATCHES=(
```

Comment:
Put stuff like this above all phases, i.e. move src_unpack down below all of this